nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -1227,105 +1210,63 @@ define(
                 });
             }
 
-            function resetToEditMode() {
+            // Reset the app to edit mode; if a new state is not supplied, it is assumed that the
+            // existing app params are complete and validated.
+            function resetToEditMode(newState) {
```

Comment:
Combined `resetAppAndEdit` into this function as they do nearly all the same things.